src/Login.jsx

```diff
@@ -1,71 +1,79 @@
-import React, { useState } from 'react';
-import './Login.css';
+import { useState } from "react";
+import "./Login.css";
+import PropTypes from "prop-types";
 
 const LoginComponent = ({ onLogin }) => {
-    const [username, setUsername] = useState('');
-    const [password, setPassword] = useState('');
-    const [error, setError] = useState('');
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
 
-    const handleLogin = async () => {
-        setError('');
+  const handleLogin = async () => {
+    setError("");
 
-        try {
-            const response = await fetch('https://reqres.in/api/login', {
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/json',
-                },
-                body: JSON.stringify({ username, password }),
-                // La API de Reqres.in tiene ciertos usuarios predefinidos para la autenticación.
-                //Usuario predefinido:eve.holt@reqres.in
-                //contraseña predefinida:cityslicka
-            });
+    if (!username && !password) {
+      setError("Por favor, ingresa un nombre de usuario y una contraseña");
+      return;
+    } else if (!username) {
+      setError("Por favor, ingresa un nombre de usuario");
+      return;
+    } else if (!password) {
+      setError("Por favor, ingresa una contraseña");
+      return;
+    }
```

Comment:
minor thing here, we can extract this handling **before** the fetch request, the idea is: you don't need to do an HTTP request if the username or the password are blank, you don't need to wait for the API response